mqtt/mqtt-bridge/src/persist/loader.rs

```diff
@@ -52,7 +54,7 @@ impl Stream for InMemoryMessageLoader {
 }
 
 fn get_elements(
-    state: &mut MutexGuard<'_, WakingMap>,
+    state: &mut MutexGuard<'_, impl StreamWakeableState>,
     batch_size: usize,
 ) -> IntoIter<(Key, Publication)> {
     let batch: Vec<_> = state
```

Comment:
you can do tuple deconstruction in closures as well `.map(|(key, publication)| (key.clone(), publication.clone()))`